src/Capture/X11.hs

```diff
@@ -60,10 +61,10 @@ captureData = do
         (fsubwin,_) <- getInputFocus dpy
         fwin <- followTreeUntil dpy (`elem` wins) fsubwin
 
-        winData <- forM wins $ \w -> (,,)
-            (w == fwin) <$>
-            (T.pack <$> getWindowTitle dpy w) <*>
-            (T.pack <$> getProgramName dpy w)
+        winData <- forM wins $ \w -> do
+            title <- T.pack <$> getWindowTitle dpy w
+            program <- T.pack <$> getProgramName dpy w
+            return def{ wActive = w == fwin, wTitle = title, wProgram = program }
```

Comment:
Did you consider using `NamedFieldPuns` here?

Also slightly worried about the `def` here.  This is a program, not a library, so it’s ok (actuall good!) to have to adjust all  places where this value is is creatd. So maybe use  a plain record literal here?

Reply:
I think I tried it and ghc told me I can't use them with def{…}. I specifically wanted to _not_ have to adjust Capture.OSX/Win32, but you're right that here it probably does make sense to use that.